src/sentry/rules/processing/delayed_processing.py

```diff
@@ -159,7 +161,17 @@ def get_rules_to_fire(
                     ):
                         for group_id in rules_to_groups[alert_rule.id]:
                             if results[group_id] > target_value:
+                                if action_match == "all":
+                                    num_conditions_passed += 1
                                 rules_to_fire[alert_rule].add(group_id)
+
+        if (
+            action_match == "all"
+            and len(slow_conditions) > 1
+            and len(slow_conditions) != num_conditions_passed
+            and rules_to_fire[alert_rule]
+        ):
+            del rules_to_fire[alert_rule]
```

Comment:
Is this correct? Can't we have some groups that pass all conditions, and some that don't?

I'm wondering if we should be iterating like this instead
```
for alert_rule, slow_conditions in rule_to_slow_conditions.items():
    for group_id in rules_to_groups[alert_rule.id]: # Are we guaranteed to have an entry for the rule here?
        conditions_matched = 0
        for slow_condition in slow_conditions: # Would be better if `slow_conditions` were `UniqueConditions` to begin with
            unique_condition = UniqueCondition(condition_id, condition_interval, alert_rule.environment_id)
            results = condition_group_results.get(unique_condition, {})
            if results[group_id] >= target_value: 
                if action_match == "any":
                    rules_to_fire[alert_rule].add(group_id)
                    break
                conditions_matched += 1
            else:
                if action_match == "all":
                    # We failed to match all conditions for this group, skip
                    break
        if action_match == "all" and conditions_matched == len(slow_conditions):
            rules_to_fire[alert_rule].add(group_id)
```